bin/edit.py

```diff
@@ -14,6 +14,10 @@
 import time
 import argparse
 
+_stash = globals()["_stash"]
+
+if _stash.PY3:
```

Comment:
@ywangd and @bennr01 Do you have any objections to us using the very popular [__six__ module](http://six.readthedocs.io) to achieve compatibility with both Python 2 and Python 3?  It will be much cleaner than a bunch of __if PY3__ statements and complex workarounds for things that are in __six.moves__.

Reply:
@cclauss no objection by me. I only decided against using `six` as i have no experience with it.

Reply:
```python
try:
    raw_input
except NameError:
    raw_input = input
```

No `six` or `PY3` required :)

Reply:
To clarify - I don't have anything against `six`, but personally I've found it overkill in most cases. The most common missing names are `unicode`, `raw_input`, and `xrange`, which can be defined when needed with a `try`-`except NameError` block. Renamed modules can be handled with `try`-`except ImportError` as well. (`urllib` and friends are perhaps the most annoying case, in that case `six` is useful.)

Also I think `six` is more necessary when you need to target ancient Python versions (older than 2.7 or 3.3). These are missing various useful 2/3 compatibility features, which `six` can emulate. 2.7 is very forward-compatible, so there aren't many things that `six` can do that you can't do yourself in a couple of lines.

Regardless of whether or not you're using `six`, `__future__` imports are very useful too. `from __future__ import absolute_import, division, print_function` is a good default for Python 2/3 code, it makes some things behave more like Python 3 even in Python 2.

Reply:
Great observations...  I tend to lean a bit more towards six than you do because it does not slow down the reader of the code but your alternatives are all correct.